scripts/ai/jobs/CpAIJobFieldWork.lua

```diff
@@ -40,15 +42,38 @@ function CpAIJobFieldWork:onPreStart()
     end
     self:addTask(self.driveToFieldWorkStartTask)
     self:addTask(self.fieldWorkTask)
+    ---TODO: only if refuelling is possible ..
+    -- self:addTask(self.refuelTask)
 end
 
 function CpAIJobFieldWork:setupJobParameters()
     CpAIJob.setupJobParameters(self)
     self:setupCpJobParameters(CpFieldWorkJobParameters(self))
 end
 
+function CpAIJobFieldWork:isFinishingAllowed(message)
```

Comment:
just a little comment on what this means? Or is this a Giants function? Also some explanation on the concept here, especially since this is again a getter function with side effects.